test/e2e/prometheus_instance_namespaces_test.go

```diff
@@ -302,11 +331,11 @@ func testPrometheusInstanceNamespacesAllowList(t *testing.T) {
 		// Remove the selecting label on the "allowed" namespace and check that
 		// the target is removed.
 		// See https://github.com/prometheus-operator/prometheus-operator/issues/3847
-		//if err := testFramework.RemoveLabelsFromNamespace(framework.KubeClient, allowedNs, "monitored"); err != nil {
+		//if err := framework.RemoveLabelsFromNamespace(context.Background(), allowedNs, "monitored"); err != nil {
```

Comment:
this part is still commented because the service account isn't configured with list/watch permissions on namespaces. I have another change in flight that would fix it by implementing a poller based watcher but to keep this PR reviewable, I'll send as a follow-up.

Reply:
https://github.com/prometheus-operator/prometheus-operator/pull/5925/commits/5ada40db1508fdb128fbd558c52bc10a1f92a33f